src/executables/sysrepo-plugind.c

```diff
@@ -323,13 +327,93 @@ load_plugins(struct srpd_plugin_s **plugins, int *plugin_count)
         /* finally store the plugin */
         (*plugins)[*plugin_count].handle = handle;
         (*plugins)[*plugin_count].private_data = NULL;
+        {
+            const char *src = ent->d_name;
+            const size_t src_len = strlen(src);
+            const size_t n = strcmp(src + (src_len - 3), ".so") == 0 ?
+                src_len - 3 : src_len;
+            char *str = strndup(src, n);
+            if (!str) {
+                error_print(0, "strndup() failed.");
+                dlclose(handle);
+                rc = -1;
+                break;
+            }
+            (*plugins)[*plugin_count].plugin_name = str;
+        }
         ++(*plugin_count);
     }
 
     closedir(dir);
     return rc;
 }
 
+static void
+swap(struct srpd_plugin_s *a, struct srpd_plugin_s *b)
+{
+    if (a == b) {
+        return;
+    }
+    struct srpd_plugin_s tmp = *a;
+    *a = *b;
+    *b = tmp;
+}
+
+static int
+plugin_names_cmp(const struct srpd_plugin_s *plugin, const char *str2)
+{
+    /* str1 does not have the .so suffix */
+    const char *str1 = plugin->plugin_name;
+    if (str1 == str2) {
+        return 0;
+    }
+    const size_t str1_len = strlen(str1);
+    const size_t str2_len = strlen(str2);
+    /* suffx .so in str2 is ignored */
+    const size_t n = strcmp(str2 + (str2_len - 3), ".so") == 0 ?
```

Comment:
To be consistent, please treat `strcmp` return value as `boolean` directly. Also, to be safe, make sure `str2_len` is at least `3`.